### DISCUSSION TITLE
v0.31.0 (Last release before v1.0)

### DISCUSSION THREAD

**Discussion by alimd** [AliMD/alwatr#1134] [Announcements]
## What's Changed

* feat(com-pwa): qty unit by @njfamirm in https://github.com/AliMD/alwatr/pull/906
* fix(icon): 404 cache issue by @AliMD in https://github.com/AliMD/alwatr/pull/911
* fix(com-pwa): review by @njfamirm in https://github.com/AliMD/alwatr/pull/910
* fix(com-pwa): color theme by @njfamirm in https://github.com/AliMD/alwatr/pull/915
* perf: make google lighthouse happy by @AliMD in https://github.com/AliMD/alwatr/pull/916
* chore(demo/es-bench): bench many func by @AliMD in https://github.com/AliMD/alwatr/pull/917
* fix(workflow): publish container signing by @njfamirm in https://github.com/AliMD/alwatr/pull/921
* perf(pwa): optimize first load and rendering by @AliMD in https://github.com/AliMD/alwatr/pull/924
* feat(ui-kit/snackbar): message/actionLabelMessage l10n key by @njfamirm in https://github.com/AliMD/alwatr/pull/926
* feat(element): ScheduleUpdateToFrameMixin by @AliMD in https://github.com/AliMD/alwatr/pull/933
* feat(bench): improve bench mark logs, test logger benchmark impact by @AliMD in https://github.com/AliMD/alwatr/pull/942
* feat(type): new type helpers by @AliMD in https://github.com/AliMD/alwatr/pull/943
* fix(pwa-helper): show user notify version by @AliMD in https://github.com/AliMD/alwatr/pull/944
* feat(signal): new RequestableContext with state by @AliMD in https://github.com/AliMD/alwatr/pull/945
* feat(icon,icon-button): issues by @AliMD in https://github.com/AliMD/alwatr/pull/946
* feat(element/mixins/signal): _addSignalListener by @AliMD in https://github.com/AliMD/alwatr/pull/950
* fix(signal): dont receivePrevious when listener is disabled by @AliMD in https://github.com/AliMD/alwatr/pull/951
* fix(logger): logMethod args type by @AliMD in https://github.com/AliMD/alwatr/pull/947
* feat(fsm): rewrite state machine by @AliMD in https://github.com/AliMD/alwatr/pull/923
* fix: compatible with new _addSignalListener signal mixin by @AliMD in https://github.com/AliMD/alwatr/pull/954
* feat(element): export lit guard by @AliMD in https://github.com/AliMD/alwatr/pull/955
* fix(fsm): initial exec actions by @AliMD in https://github.com/AliMD/alwatr/pull/956
* refactor(com/order-list): with new FSM api by @AliMD in https://github.com/AliMD/alwatr/pull/948
* feat(fetch): enhance types and add StringifyableFetchOptions by @AliMD in https://github.com/AliMD/alwatr/pull/959
* feat(fsm): add destroy, reset, unsubscribe api by @AliMD in https://github.com/AliMD/alwatr/pull/960
* refactor(type): AlwatrDocumentStorage type with AlwatrServiceResponseSuccessWithMeta by @AliMD in https://github.com/AliMD/alwatr/pull/961
* feat(context): new package for server context over `FSM` and `context signal` <fucking awesome> by @AliMD in https://github.com/AliMD/alwatr/pull/958
* refactor(context): review and enhance core api by @AliMD in https://github.com/AliMD/alwatr/pull/963
* refactor(SignalMixin): rename _addSignalListeners by @AliMD in https://github.com/AliMD/alwatr/pull/964
* fix(context): merge request option issue by @AliMD in https://github.com/AliMD/alwatr/pull/965
* feat(com/order-list): update order list with new `fsm` and new `context` ;) by @AliMD in https://github.com/AliMD/alwatr/pull/962
* refactor(soffit-pwa): move l18e to content folder by @njfamirm in https://github.com/AliMD/alwatr/pull/922
* perf(pwa): optimize index.html by @njfamirm in https://github.com/AliMD/alwatr/pull/996
* fix(pwa-helper/esbuild): hash entryNames by @njfamirm in https://github.com/AliMD/alwatr/pull/995
* chore(workflow): enable for all pull request by @njfamirm in https://github.com/AliMD/alwatr/pull/992
* feat(logger): add time and timeEnd by @AliMD in https://github.com/AliMD/alwatr/pull/1002
* chore: new es benchs by @AliMD in https://github.com/AliMD/alwatr/pull/1003
* feat(com-pwa): refactor order-detail and new-order with new `FSM` and `context` api by @njfamirm in https://github.com/AliMD/alwatr/pull/957
* feat(com): cdn by @njfamirm in https://github.com/AliMD/alwatr/pull/983
* feat(button): use content property and move to manager by @njfamirm in https://github.com/AliMD/alwatr/pull/978
* feat(cloud/com): add chat service by @njfamirm in https://github.com/AliMD/alwatr/pull/993
* feat(com-pwa): chat storage context provider by @njfamirm in https://github.com/AliMD/alwatr/pull/994
* chore: pwa version by @njfamirm in https://github.com/AliMD/alwatr/pull/1008
* fix(ui/text-field): set empty string for undefined value by @njfamirm in https://github.com/AliMD/alwatr/pull/1009
* feat(com-pwa): change order_shipping_car i18n key by @njfamirm in https://github.com/AliMD/alwatr/pull/1016
* feat(nano-server): prefix pattern by @njfamirm in https://github.com/AliMD/alwatr/pull/1010
* fix(nano-server): prefixPattern, logger by @AliMD in https://github.com/AliMD/alwatr/pull/1018
* feat(ui/icon): svg property by @MM25Zamanian in https://github.com/AliMD/alwatr/pull/867
* fix(cloud): exclude deploy backup by @AliMD in https://github.com/AliMD/alwatr/pull/1019
* refactor(logger)!: new api by @AliMD in https://github.com/AliMD/alwatr/pull/1017
* docs(fsm): docs methods by @AliMD in https://github.com/AliMD/alwatr/pull/1021
* fix(fetch): compatible with node by @njfamirm in https://github.com/AliMD/alwatr/pull/1045
* fix(telegram-notifier): escaping send message by @AliMD in https://github.com/AliMD/alwatr/pull/1049
* chore(cloud): update env by @AliMD in https://github.com/AliMD/alwatr/pull/1048
* feat(com-pwa): review, refactor, release v1 by @AliMD in https://github.com/AliMD/alwatr/pull/1050
* feat(nginx): optimize pwa spa serve by @AliMD in https://github.com/AliMD/alwatr/pull/1057
* feat(nginx): optimize base for cdn and enhance performance by @AliMD in https://github.com/AliMD/alwatr/pull/1064
* feat(nginx-pwa): new nginx container, optimized for pwa by @AliMD in https://github.com/AliMD/alwatr/pull/1065
* feat(nginx-storage): new nginx container, optimized for storage by @AliMD in https://github.com/AliMD/alwatr/pull/1066
* fix(cloud/classic): update `env`s by @AliMD in https://github.com/AliMD/alwatr/pull/1067
* feat(container/nginx-storage): error.json for error by @njfamirm in https://github.com/AliMD/alwatr/pull/1076
* feat(com-pwa): new sign-in page and auth process by @njfamirm in https://github.com/AliMD/alwatr/pull/1060
* feat(com-api): update price list by @njfamirm in https://github.com/AliMD/alwatr/pull/1035
* fix(nginx): NGINX_ACCESS_LOG default value by @AliMD in https://github.com/AliMD/alwatr/pull/1078
* fix(nginx-storage): ssi syntax error by @AliMD in https://github.com/AliMD/alwatr/pull/1079
* fix(nginx-storage): ssi types by @njfamirm in https://github.com/AliMD/alwatr/pull/1080
* feat(validator): sanitizePhoneNumber by @AliMD in https://github.com/AliMD/alwatr/pull/1086
* chore(com-api): add new product by @njfamirm in https://github.com/AliMD/alwatr/pull/1085
* feat(com-pwa): complete sign in form by @njfamirm in https://github.com/AliMD/alwatr/pull/1077
* feat(util): setLocalStorageItem by @njfamirm in https://github.com/AliMD/alwatr/pull/1092
* feat(type/user): add `lpe` by @AliMD in https://github.com/AliMD/alwatr/pull/1097
* feat(crypto): self validate hash and user factory by @AliMD in https://github.com/AliMD/alwatr/pull/1096
* feat(crypto): rename package & demo by @njfamirm in https://github.com/AliMD/alwatr/pull/1098
* feat(crypto): make prefix for userId by @njfamirm in https://github.com/AliMD/alwatr/pull/1099
* feat(com-api): new storage nginx server and signup process by @njfamirm in https://github.com/AliMD/alwatr/pull/1100
* feat(com-pwa): new preview and fix reported issues by @njfamirm in https://github.com/AliMD/alwatr/pull/1081
* feat(i18n): date method for convert date to active locale by @njfamirm in https://github.com/AliMD/alwatr/pull/1033
* feat(com-pwa): order created date in status box by @njfamirm in https://github.com/AliMD/alwatr/pull/1034
* Feat/shipping info by @njfamirm in https://github.com/AliMD/alwatr/pull/1103
* fix(com-pwa): reported issue and deploy canary and beta by @njfamirm in https://github.com/AliMD/alwatr/pull/1104
* refactor(com-pwa): order storage path by @njfamirm in https://github.com/AliMD/alwatr/pull/1109
* feat(nginx): enhance and dynamic request limit rate by @AliMD in https://github.com/AliMD/alwatr/pull/1112
* fix(soffit-order): reported issues by @AliMD in https://github.com/AliMD/alwatr/pull/1113
* feat(cloud): mashin-ro wp by @njfamirm in https://github.com/AliMD/alwatr/pull/1110
* feat(type): UserAuth and user.permissions by @AliMD in https://github.com/AliMD/alwatr/pull/1120
* feat(fetch, nano-server): user/auth by @AliMD in https://github.com/AliMD/alwatr/pull/1121
* feat(storage-client, server): touch method by @AliMD in https://github.com/AliMD/alwatr/pull/1122
* feat(util): node read write json helper methods by @AliMD in https://github.com/AliMD/alwatr/pull/1123
* feat(logger): add `duration time` to logger time label by @AliMD in https://github.com/AliMD/alwatr/pull/1125
* feat(storage-engine): none blocking save storage by @AliMD in https://github.com/AliMD/alwatr/pull/1124
* feat(util/node): makeLinkForce by @njfamirm in https://github.com/AliMD/alwatr/pull/1128
* feat(storage-server/client): link method by @njfamirm in https://github.com/AliMD/alwatr/pull/1126
* feat(storage-client/server): cacheApiResponse method by @AliMD in https://github.com/AliMD/alwatr/pull/1129
* feat(nginx): add NGINX_DISABLE_SYMLINKS env and fix rate issue by @AliMD in https://github.com/AliMD/alwatr/pull/1130
* feat(pwa-helper): add build short sha to alwatr version by @AliMD in https://github.com/AliMD/alwatr/pull/1131
* fix(workflow/pwa): add git to docker build by @njfamirm in https://github.com/AliMD/alwatr/pull/1132

* chore(deps-dev): bump esbuild from 0.17.10 to 0.17.11 by @dependabot in https://github.com/AliMD/alwatr/pull/909
* chore(deps-dev): bump eslint from 8.35.0 to 8.36.0 by @dependabot in https://github.com/AliMD/alwatr/pull/931
* chore(deps-dev): bump typescript from 4.9.5 to 5.0.2 by @dependabot in https://github.com/AliMD/alwatr/pull/953
* chore(deps-dev): bump esbuild from 0.17.11 to 0.17.12 by @dependabot in https://github.com/AliMD/alwatr/pull/952
* chore(deps): bump zx from 7.2.0 to 7.2.1 by @dependabot in https://github.com/AliMD/alwatr/pull/949
* chore(deps): bump actions/checkout from 3.3.0 to 3.4.0 by @dependabot in https://github.com/AliMD/alwatr/pull/940
* chore(deps-dev): bump @web/dev-server from 0.1.35 to 0.1.36 by @dependabot in https://github.com/AliMD/alwatr/pull/939
* chore(deps): bump github/codeql-action from 2.2.5 to 2.2.7 by @dependabot in https://github.com/AliMD/alwatr/pull/938
* chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.54.0 to 5.55.0 by @dependabot in https://github.com/AliMD/alwatr/pull/937
* chore(deps): bump docker/setup-buildx-action from 2.4.1 to 2.5.0 by @dependabot in https://github.com/AliMD/alwatr/pull/928
* chore(deps): bump telegraf from 4.11.2 to 4.12.2 by @dependabot in https://github.com/AliMD/alwatr/pull/927
* chore(deps-dev): bump @material/web from 1.0.0-pre.3 to 1.0.0-pre.4 by @dependabot in https://github.com/AliMD/alwatr/pull/925
* chore(deps-dev): bump @types/node from 18.14.5 to 18.15.3 by @dependabot in https://github.com/AliMD/alwatr/pull/936
* chore(deps-dev): bump @typescript-eslint/parser from 5.54.0 to 5.55.0 by @dependabot in https://github.com/AliMD/alwatr/pull/935
* chore(deps): bump lit from 2.6.1 to 2.7.0 by @dependabot in https://github.com/AliMD/alwatr/pull/973
* chore(deps-dev): bump @types/node from 18.15.3 to 18.15.5 by @dependabot in https://github.com/AliMD/alwatr/pull/972
* chore(deps-dev): bump @typescript-eslint/parser from 5.55.0 to 5.56.0 by @dependabot in https://github.com/AliMD/alwatr/pull/971
* chore(deps-dev): bump prettier from 2.8.4 to 2.8.6 by @dependabot in https://github.com/AliMD/alwatr/pull/969
* chore(deps): bump actions/dependency-review-action from 3.0.3 to 3.0.4 by @dependabot in https://github.com/AliMD/alwatr/pull/968
* chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.55.0 to 5.56.0 by @dependabot in https://github.com/AliMD/alwatr/pull/970
* chore(deps-dev): bump eslint-import-resolver-typescript from 3.5.3 to 3.5.4 by @dependabot in https://github.com/AliMD/alwatr/pull/1001
* chore(deps-dev): bump eslint from 8.36.0 to 8.37.0 by @dependabot in https://github.com/AliMD/alwatr/pull/999
* chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.56.0 to 5.57.0 by @dependabot in https://github.com/AliMD/alwatr/pull/989
* chore(deps-dev): bump esbuild from 0.17.12 to 0.17.14 by @dependabot in https://github.com/AliMD/alwatr/pull/986
* chore(deps-dev): bump lerna from 6.5.1 to 6.6.1 by @dependabot in https://github.com/AliMD/alwatr/pull/985
* chore(deps): bump actions/checkout from 3.4.0 to 3.5.0 by @dependabot in https://github.com/AliMD/alwatr/pull/982
* chore(deps-dev): bump prettier from 2.8.6 to 2.8.7 by @dependabot in https://github.com/AliMD/alwatr/pull/979
* chore(deps-dev): bump nodemon from 2.0.21 to 2.0.22 by @dependabot in https://github.com/AliMD/alwatr/pull/976
* chore(deps): bump actions/stale from 7.0.0 to 8.0.0 by @dependabot in https://github.com/AliMD/alwatr/pull/974
* chore(deps-dev): bump @web/dev-server from 0.1.36 to 0.1.37 by @dependabot in https://github.com/AliMD/alwatr/pull/997
* chore(deps-dev): bump @types/node from 18.15.5 to 18.15.11 by @dependabot in https://github.com/AliMD/alwatr/pull/998
* chore(deps): bump github/codeql-action from 2.2.7 to 2.2.9 by @dependabot in https://github.com/AliMD/alwatr/pull/990
* chore(deps-dev): bump @material/web from 1.0.0-pre.4 to 1.0.0-pre.5 by @dependabot in https://github.com/AliMD/alwatr/pull/977
* chore(deps-dev): bump @typescript-eslint/parser from 5.56.0 to 5.57.0 by @dependabot in https://github.com/AliMD/alwatr/pull/988
* chore(deps): bump lit from 2.7.0 to 2.7.1 by @dependabot in https://github.com/AliMD/alwatr/pull/1014
* chore(deps-dev): bump esbuild from 0.17.14 to 0.17.15 by @dependabot in https://github.com/AliMD/alwatr/pull/1011
* chore(deps-dev): bump typescript from 5.0.2 to 5.0.3 by @dependabot in https://github.com/AliMD/alwatr/pull/1004
* chore(deps-dev): bump @web/dev-server from 0.1.37 to 0.1.38 by @dependabot in https://github.com/AliMD/alwatr/pull/1020
* chore(deps-dev): bump esbuild from 0.17.15 to 0.17.16 by @dependabot in https://github.com/AliMD/alwatr/pull/1039
* chore(deps-dev): bump eslint from 8.37.0 to 8.38.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1038
* chore(deps-dev): bump typescript from 5.0.3 to 5.0.4 by @dependabot in https://github.com/AliMD/alwatr/pull/1037
* chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.57.0 to 5.59.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1054
* chore(deps-dev): bump eslint-plugin-lit from 1.8.2 to 1.8.3 by @dependabot in https://github.com/AliMD/alwatr/pull/1056
* chore(deps-dev): bump esbuild from 0.17.16 to 0.17.17 by @dependabot in https://github.com/AliMD/alwatr/pull/1051
* chore(deps-dev): bump @typescript-eslint/parser from 5.57.0 to 5.59.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1053
* chore(deps): bump lit from 2.7.1 to 2.7.2 by @dependabot in https://github.com/AliMD/alwatr/pull/1022
* chore(deps-dev): bump @web/dev-server from 0.1.38 to 0.2.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1063
* chore(deps-dev): bump @types/node from 18.15.11 to 18.15.13 by @dependabot in https://github.com/AliMD/alwatr/pull/1062
* chore(deps): bump docker/metadata-action from 4.3.0 to 4.4.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1055
* chore(deps): bump actions/checkout from 3.5.0 to 3.5.2 by @dependabot in https://github.com/AliMD/alwatr/pull/1047
* chore(deps-dev): bump @material/web from 1.0.0-pre.5 to 1.0.0-pre.6 by @dependabot in https://github.com/AliMD/alwatr/pull/1042
* chore(deps): bump sigstore/cosign-installer from 3.0.1 to 3.0.2 by @dependabot in https://github.com/AliMD/alwatr/pull/1036
* chore(deps): bump github/codeql-action from 2.2.9 to 2.3.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1070
* chore(deps-dev): bump @types/node from 18.15.13 to 18.16.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1069
* chore(deps-dev): bump eslint-import-resolver-typescript from 3.5.4 to 3.5.5 by @dependabot in https://github.com/AliMD/alwatr/pull/1023
* chore(deps): bump actions/add-to-project from 0.4.1 to 0.5.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1013
* chore(deps-dev): bump @webcomponents/webcomponentsjs from 2.7.0 to 2.8.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1005
* chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.59.0 to 5.59.1 by @dependabot in https://github.com/AliMD/alwatr/pull/1083
* chore(deps-dev): bump eslint from 8.38.0 to 8.39.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1075
* chore(deps-dev): bump prettier from 2.8.7 to 2.8.8 by @dependabot in https://github.com/AliMD/alwatr/pull/1074
* chore(deps-dev): bump esbuild from 0.17.17 to 0.17.18 by @dependabot in https://github.com/AliMD/alwatr/pull/1073
* chore(deps-dev): bump eslint-plugin-wc from 1.4.0 to 1.5.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1072
* chore(deps-dev): bump @web/dev-server from 0.2.0 to 0.2.1 by @dependabot in https://github.com/AliMD/alwatr/pull/1071
* chore(deps-dev): bump @typescript-eslint/parser from 5.59.0 to 5.59.1 by @dependabot in https://github.com/AliMD/alwatr/pull/1082
* chore(deps-dev): bump @material/web from 1.0.0-pre.6 to 1.0.0-pre.7 by @dependabot in https://github.com/AliMD/alwatr/pull/1084
* chore(deps): bump lit from 2.7.2 to 2.7.3 by @dependabot in https://github.com/AliMD/alwatr/pull/1089
* chore(deps): bump sigstore/cosign-installer from 3.0.2 to 3.0.3 by @dependabot in https://github.com/AliMD/alwatr/pull/1090
* chore(deps-dev): bump @types/node from 18.16.0 to 18.16.1 by @dependabot in https://github.com/AliMD/alwatr/pull/1088
* chore(deps-dev): bump @types/node from 18.16.1 to 18.16.2 by @dependabot in https://github.com/AliMD/alwatr/pull/1094
* chore(deps): bump lit from 2.7.3 to 2.7.4 by @dependabot in https://github.com/AliMD/alwatr/pull/1117
* chore(deps): bump github/codeql-action from 2.3.0 to 2.3.3 by @dependabot in https://github.com/AliMD/alwatr/pull/1116
* chore(deps-dev): bump lerna from 6.6.1 to 6.6.2 by @dependabot in https://github.com/AliMD/alwatr/pull/1115
* chore(deps-dev): bump @types/node from 18.16.2 to 20.1.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1119
* chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.59.1 to 5.59.2 by @dependabot in https://github.com/AliMD/alwatr/pull/1107
* chore(deps-dev): bump @typescript-eslint/parser from 5.59.1 to 5.59.2 by @dependabot in https://github.com/AliMD/alwatr/pull/1106
* chore(deps): bump zx from 7.2.1 to 7.2.2 by @dependabot in https://github.com/AliMD/alwatr/pull/1108
* chore(deps-dev): bump eslint from 8.39.0 to 8.40.0 by @dependabot in https://github.com/AliMD/alwatr/pull/1127

**Full Changelog**: https://github.com/AliMD/alwatr/compare/v0.30.0...v0.31.0

<hr /><em>This discussion was created from the release <a href='https://github.com/AliMD/alwatr/releases/tag/v0.31.0'>v0.31.0</a>.</em>

**Comment by njfamirm** [comment]
ریلیزی که کلش فورس بودیم برای همه کاراش 🥵
